core/tests/test_utils.py

```diff
@@ -3673,7 +3673,9 @@ def __call__(self, *args, **kwargs):
         if self._instance is not None:
             args = [self._instance] + list(args)
 
-        args_dict = inspect.getcallargs(self._func, *args, **kwargs)
+        sig = inspect.signature(self._func)
+
+        args_dict = sig.bind_partial(*args, **kwargs)
```

Comment:
Why this change?

Reply:
getcallargs() was also deprecated in the new python version and was showing the same error as other deprecated methods when pylint ran.
![Screenshot from 2021-12-29 22-59-19](https://user-images.githubusercontent.com/89375125/147695574-99f4736d-065b-4935-b300-d2ff8da1f84e.png)

Reply:
Okay, could you then do 
```suggestion
        sig = inspect.signature(self._func)
        args_dict = sig.bind_partial(*args, **kwargs)
```

and add an explanatory comment above it explaining what this code does.

Reply:
Done!

Reply:
Sorry, but I made a mistake here as this would not be the correct way to replace inspect.getcallargs( ). The method getcallargs() returns a dict while bind_partial( ) returns BoundArguments. This [test](https://github.com/oppia/oppia/runs/4690325462?check_suite_focus=true#step:8:1719) is failing due to this issue.

I searched for the valid solution of this issue but could not find any satisfactory solution except these lines of code which use bind( ) to return dict.
![Screenshot from 2022-01-03 17-11-57](https://user-images.githubusercontent.com/89375125/147926894-46c5e344-1d4c-4f16-a301-541df3a35c53.png)

However I am not sure of this and open to suggestions.
This error didn't show up in my localmachine. I am sorry if I caused any troubles.

Reply:
@vojtechjelinek PTAL

Reply:
@vojtechjelinek Can you please help here?

Reply:
@IamEzio Hey look at this https://docs.python.org/3/library/inspect.html#inspect.BoundArguments, the structure of `BoundArguments` is explained there, from this you should be able to change the code.